### PR TITLE
Fix remaining lint errors in `build-system` and enforce lint checks via Travis

### LINKED ISSUE #11273: Add lint rules for `build-system/`
I just took the existing linter that we use for the AMP runtime and ran it against `build-system/` (minus `build-system/node-modules/`), and encountered numerous errors.

https://gist.github.com/rsimha-amp/c9e5e1caa4829bd65ea94c373068993a

Several errors are due to the fact that we use ES5 style for the build system while the lint rules are for ES6. However, there are a significant number of genuine errors that should gradually be found and fixed.

### PR DESCRIPTION
**Pull request by rsimha-amp**
Follow up to #11325
Follow up to #11334
Follow up to #11357
Fixes #11273

### CODE REVIEW

File: build-system/amp.extern.js [hunk 1]
```diff
@@ -14,6 +14,8 @@
  * limitations under the License.
  */
```
Comment (jridgewell) [inline build-system/amp.extern.js]: We shouldn't be linting this file.

File: build-system/amp.nti.extern.js [hunk 2]
```diff
@@ -27,4 +27,4 @@
  * @return {function(number):T}
  * @template T
  */
-var TransitionDef = function(unusedNormtime, unusedCompleted) {};
+const TransitionDef = function(unusedNormtime, unusedCompleted) {};
```
Comment (jridgewell) [inline build-system/amp.nti.extern.js]: Shouldn't be linting.

File: build-system/amp.oti.extern.js [hunk 3]
```diff
@@ -24,4 +24,4 @@
  * transition and "false" for ongoing.
  * @typedef {function(number, boolean):?|function(number):?}
  */
-var TransitionDef;
+let TransitionDef;
```
Comment (jridgewell) [inline build-system/amp.oti.extern.js]: Shouldn't be linting.

File: build-system/app.js [hunk 4]
```diff
@@ -1199,4 +1198,4 @@ function generateInfo(filePath) {
       '<h3><a href = /serve_mode=cdn>Change to CDN mode (prod JS)</a></h3>';
 }
 
-module.exports = app;
+module.exports = app;  // eslint-disable-line no-undef
```
Comment (jridgewell) [inline build-system/app.js]: What's undefined?

File: build-system/config.js [hunk 5]
```diff
@@ -56,50 +56,47 @@ var commonTestPaths = [
...
-  testPaths: testPaths,
-  a4aTestPaths: a4aTestPaths,
-  unitTestPaths: unitTestPaths,
-  integrationTestPaths: integrationTestPaths,
-  buildSystemLintGlobs: [
-    'build-system/tasks/**/*.js',
-    '!build-system/eslint-rules/**/*.*',
-    '!build-system/node_modules/**/*.*',
-  ],
+module.exports = {  // eslint-disable-line no-undef
```
Comment (jridgewell) [inline build-system/config.js]: What's undefined?

File: build-system/pr-check.js [hunk 6]
```diff
@@ -139,7 +127,7 @@ function isBuildSystemFile(filePath) {
  * @return {boolean}
  */
 function isValidatorFile(filePath) {
-  if (filePath.startsWith('validator/')) return true;
+  if (filePath.startsWith('validator/')) {return true;}
```
Comment (jridgewell) [inline build-system/pr-check.js]: New line

File: build-system/tasks/changelog.js [hunk 7]
```diff
@@ -536,7 +536,7 @@ function buildPrMetadata(pr) {
     id: pr.number,
     title: pr.title,
     body: pr.body,
-    merge_commit_sha: pr.merge_commit_sha,
+    merge_commit_sha: pr.merge_commit_sha,  // eslint-disable-line google-camelcase/google-camelcase
```
Comment (jridgewell) [inline build-system/tasks/changelog.js]: This should use quotes to avoid.

File: build-system/tasks/compile.js [hunk 8]
```diff
@@ -86,7 +86,7 @@ exports.cleanupBuildDir = cleanupBuildDir;
 
 function compile(entryModuleFilenames, outputDir,
     outputFilename, options) {
-  return new Promise(function(resolve, reject) {
+  return new Promise(function(resolve, unusedReject) {
```
Comment (jridgewell) [inline build-system/tasks/compile.js]: Remove.

File: build-system/tasks/csvify-size/index.js [hunk 9]
```diff
@@ -29,7 +29,7 @@ const prettyBytesUnits = ['B', 'kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
 /**
  * @typedef {!Array<Fields>}
  */
-let Tables;
+let Tables;  // eslint-disable-line no-unused-vars
```
Comment (jridgewell) [inline build-system/tasks/csvify-size/index.js]: Rename to `TablesDef`.

File: build-system/tasks/csvify-size/index.js [hunk 10]
```diff
@@ -38,12 +38,10 @@ let Tables;
  *   size: string
  * }}
  */
-let Fields;
+let Fields;  // eslint-disable-line no-unused-vars
```
Comment (jridgewell) [inline build-system/tasks/csvify-size/index.js]: `FieldsDef`

File: build-system/tasks/dep-check.js [hunk 11]
```diff
@@ -223,7 +219,7 @@ function flattenGraph(entryPoints) {
   entryPoints = entryPoints.map(entryPoint => entryPoint.deps);
   // Now make the graph have unique entries
   return flatten(entryPoints)
-      .reduce((acc, cur, i, arr) => {
+      .reduce((acc, cur, unusedI, unusedArr) => {
```
Comment (jridgewell) [inline build-system/tasks/dep-check.js]: Remove.

File: build-system/tasks/gen-codeowners/index.js [hunk 12]
```diff
@@ -92,7 +91,7 @@ function generate(root, target, writeToDisk) {
         const dirname = path.relative(process.cwd(), path.dirname(file.path));
         dirs[dirname || '*'] = yaml.parse(file.contents.toString());
       }))
-      .on('end', function(cb) {
+      .on('end', function(unusedCb) {
```
Comment (jridgewell) [inline build-system/tasks/gen-codeowners/index.js]: Remove.

File: build-system/tasks/get-zindex/index.js [hunk 13]
```diff
@@ -76,7 +76,7 @@ function onFileThrough(file, enc, cb) {
   postcss([zIndexCollector.bind(null, selectors)])
       .process(file.contents.toString(), {
         from: file.relative,
-      }).then(res => {
+      }).then(unusedRes => {
```
Comment (jridgewell) [inline build-system/tasks/get-zindex/index.js]: Remove.

File: build-system/tasks/karma.conf.js [hunk 14]
```diff
@@ -18,7 +18,7 @@
 /**
  * @param {!Object} config
  */
-module.exports = {
+module.exports = {  // eslint-disable-line no-undef
```
Comment (jridgewell) [inline build-system/tasks/karma.conf.js]: We should set the `eslintrc` to `node` environment.

File: build-system/app.js (hunk 4 above)
Comment (rsimha-amp) [reply build-system/app.js]: Linter complains that `module` is undefined.

File: build-system/tasks/changelog.js (hunk 7 above)
Comment (rsimha-amp) [reply build-system/tasks/changelog.js]: I tried quotes. The error is in `pr.merge_commit_sha`, which appears to be a well defined github constant.

File: build-system/amp.extern.js (hunk 1 above)
Comment (rsimha-amp) [reply build-system/amp.extern.js]: Reverted.

File: build-system/amp.nti.extern.js (hunk 2 above)
Comment (rsimha-amp) [reply build-system/amp.nti.extern.js]: Reverted.

File: build-system/amp.oti.extern.js (hunk 3 above)
Comment (rsimha-amp) [reply build-system/amp.oti.extern.js]: Reverted.

File: build-system/config.js (hunk 5 above)
Comment (rsimha-amp) [reply build-system/config.js]: `module`.

File: build-system/pr-check.js (hunk 6 above)
Comment (rsimha-amp) [reply build-system/pr-check.js]: Done.

File: build-system/tasks/compile.js (hunk 8 above)
Comment (rsimha-amp) [reply build-system/tasks/compile.js]: Done.

File: build-system/tasks/csvify-size/index.js (hunk 9 above)
Comment (rsimha-amp) [reply build-system/tasks/csvify-size/index.js]: Done.

File: build-system/tasks/csvify-size/index.js (hunk 10 above)
Comment (rsimha-amp) [reply build-system/tasks/csvify-size/index.js]: Done.

File: build-system/tasks/dep-check.js (hunk 11 above)
Comment (rsimha-amp) [reply build-system/tasks/dep-check.js]: Removed.

File: build-system/tasks/gen-codeowners/index.js (hunk 12 above)
Comment (rsimha-amp) [reply build-system/tasks/gen-codeowners/index.js]: Done.

File: build-system/tasks/get-zindex/index.js (hunk 13 above)
Comment (rsimha-amp) [reply build-system/tasks/get-zindex/index.js]: Done.

File: build-system/tasks/karma.conf.js (hunk 14 above)
Comment (rsimha-amp) [reply build-system/tasks/karma.conf.js]: Done.

File: build-system/app.js (hunk 4 above)
Comment (rsimha-amp) [reply build-system/app.js]: Removed after setting `env` to `node`.

File: build-system/config.js (hunk 5 above)
Comment (rsimha-amp) [reply build-system/config.js]: Removed after setting `env` to `node`.

File: build-system/tasks/changelog.js (hunk 7 above)
Comment (jridgewell) [reply build-system/tasks/changelog.js]: Both can use quotes.

File: build-system/tasks/get-zindex/index.js [hunk 15]
```diff
@@ -87,11 +87,11 @@ function onFileThrough(file, enc, cb) {
  * @param {function()} cb callback to end the stream
  * @return {!Array<!Array<string>>}
  */
-function createTable(filesData, cb) {
+function createTable(filesData, unusedCb) {
```
Comment (jridgewell) [inline build-system/tasks/get-zindex/index.js]: Remove anything that starts with "unused".

File: build-system/tasks/json-check.js [hunk 16]
```diff
@@ -26,7 +26,7 @@ const through2 = require('through2');
 function checkValidJson() {
   let hasError = false;
   return gulp.src(jsonGlobs)
-      .pipe(through2.obj(function(file, enc, cb) {
+      .pipe(through2.obj(function(file, unusedEnc, unusedCb) {
```
Comment (jridgewell) [inline build-system/tasks/json-check.js]: Remove.

File: build-system/tasks/make-golden.js [hunk 17]
```diff
@@ -204,7 +204,7 @@ function diffScreenshot_(file, dir, host, verbose, cb) {
           util.log(util.colors.red('Screenshot diff failed: ', file, error));
           cb({error});
         })
-        .on('end', function(res) {
+        .on('end', function(unusedRes) {
```
Comment (jridgewell) [inline build-system/tasks/make-golden.js]: Remove.

File: build-system/tasks/release-tagging.js [hunk 18]
```diff
@@ -155,7 +150,7 @@ function gitFetch(dir) {
       args: 'clone https://github.com/ampproject/amphtml.git',
     });
   }
-  return clonePromise.then(function(arg) {
+  return clonePromise.then(function(unusedArg) {
```
Comment (jridgewell) [inline build-system/tasks/release-tagging.js]: Remove.

File: build-system/tasks/csvify-size/index.js [hunk 19]
```diff
@@ -164,12 +161,13 @@ function mergeTables(dateTimes, tables) {
   // populate all other columns with their respective file size if any.
   dateTimes.forEach(dateTime => {
     // Seed array with empty string values
-    const row = Array.apply(null, Array(tableHeaders[0].length)).map(x => '""');
+    const row =
+        Array.apply(null, Array(tableHeaders[0].length)).map(unusedX => '""');
```
Comment (rsimha-amp) [inline build-system/tasks/csvify-size/index.js]: Removed.

File: build-system/tasks/json-check.js (hunk 16 above)
Comment (rsimha-amp) [reply build-system/tasks/json-check.js]: Removed.

File: build-system/tasks/get-zindex/index.js (hunk 15 above)
Comment (rsimha-amp) [reply build-system/tasks/get-zindex/index.js]: Scanned the whole PR, removed all others.

File: build-system/tasks/make-golden.js (hunk 17 above)
Comment (rsimha-amp) [reply build-system/tasks/make-golden.js]: Removed.

File: build-system/tasks/release-tagging.js (hunk 18 above)
Comment (rsimha-amp) [reply build-system/tasks/release-tagging.js]: Removed.

File: build-system/tasks/changelog.js (hunk 7 above)
Comment (rsimha-amp) [reply build-system/tasks/changelog.js]: Just the field names needed quotes. The rest is fixed with `env` being set to `node`.

**Comment by rsimha-amp** [comment]
@choumx @jridgewell Can I bug one of you to take another look at this? I'd like to merge this soon since any conflicting changes that get merged before this will be a pain to resolve.

File: build-system/app.js [hunk 20]
```diff
@@ -313,7 +312,7 @@ function proxyToAmpProxy(req, res, mode) {
     body = body
         // Unversion URLs.
         .replace(/https\:\/\/cdn\.ampproject\.org\/rtv\/\d+\//g,
-            'https://cdn.ampproject.org/')
+        'https://cdn.ampproject.org/')
```
Comment (choumx) [inline build-system/app.js]: Is this indentation right?

Comment (rsimha-amp) [reply build-system/app.js]: Per our existing rules, it is right. First continuation line is indented by 4 spaces, following lines use the same indentation. This was done automatically by `--fix`, and wasn't a manual change. If this seems wrong to you, I could look at adding a more refined rule and rerunning the linter in `--fix` mode across all our files.

**Pull request merged**